having trouble running CSP version

Hi! We retrained the model with the CSP settings, using MP-20 to start. It seems to have finished training successfully. But we can't figure out how to run the generate commands, though. We tried this:
`mattergen-generate $RESULTS_PATH --model_path=$MODEL_PATH --sampling-config-name=csp --target_compositions=[{"Na": 1, "Cl": 1}] --batch_size=16 --num_batches ` 
with `$MODEL_PATH` set to `outputs/singlerun/2025-03-19/12-11-18`

the checkpoint files are in there, but what I get is:
```
(errors)
...
File "/home/fas/MATTERGEN/mattergen/mattergen/scripts/generate.py", line 55, in main
    pretrained_name is None or model_path is None
```
I tried using the path to the checkpoint file itself, which didn't help. Any tips?

Sorry,`num_batches` was set to 1, it just got cut off above in the copy-paste.

Here is the full error:
```
mattergen-generate $RESULTS_PATH --model_path=$MODEL_PATH --target_compositions=[{"Na": 1, "Cl": 1}] --batch_size=16 --num_batches 1
MODELS_PROJECT_ROOT: /home/fas/MATTERGEN/mattergen/mattergen
Traceback (most recent call last):
  File "/home/fas/MATTERGEN/.venv/bin/mattergen-generate", line 10, in <module>
    sys.exit(_main())
  File "/home/fas/MATTERGEN/mattergen/mattergen/scripts/generate.py", line 102, in _main
    fire.Fire(main)
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 135, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 468, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 684, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/fas/MATTERGEN/mattergen/mattergen/scripts/generate.py", line 55, in main
    pretrained_name is None or model_path is None
AssertionError: Only one of pretrained_name or model_path can be provided.
(.venv) 1 fas@milan2:~/MATTERGEN/mattergen$ export MODEL_PATH=../Mark-mattergen/mattergen/outputs/singlerun/2025-03-19/12-11-18/checkpoints/epoch=899-step=48600.ckpt
(.venv) fas@milan2:~/MATTERGEN/mattergen$ mattergen-generate $RESULTS_PATH --model_path=$MODEL_PATH --target_compositions=[{"Na": 1, "Cl": 1}] --batch_size=16 --num_batches 1
MODELS_PROJECT_ROOT: /home/fas/MATTERGEN/mattergen/mattergen
Traceback (most recent call last):
  File "/home/fas/MATTERGEN/.venv/bin/mattergen-generate", line 10, in <module>
    sys.exit(_main())
  File "/home/fas/MATTERGEN/mattergen/mattergen/scripts/generate.py", line 102, in _main
    fire.Fire(main)
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 135, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 468, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/fas/MATTERGEN/.venv/lib/python3.10/site-packages/fire/core.py", line 684, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/fas/MATTERGEN/mattergen/mattergen/scripts/generate.py", line 55, in main
    pretrained_name is None or model_path is None
AssertionError: Only one of pretrained_name or model_path can be provided.
```

--- a/README.md
+++ b/README.md
@@ -245,8 +245,8 @@ mattergen-train data_module=alex_mp_20 ~trainer.logger trainer.accumulate_grad_b
 Even though not a focus of our paper, you can also train MatterGen in crystal structure prediction (CSP) mode, where it does not denoise the atom types during generation. 
 This gives you the ability to condition on a specific chemical formula for generation. You can train MatterGen in this mode by passing `--config-name=csp` to `run.py`.
 
-To sample from this model, pass `--target_compositions=[{"<element1>": <number_of_element1_atoms>, "<element2>": <number_of_element2_atoms>, ..., "<elementN>": <number_of_elementN_atoms>}] --sampling-config-name=csp` to `generate.py`. 
-An example composition could be `--target_compositions=[{"Na": 1, "Cl": 1}]`.
+To sample from this model, pass `--target_compositions=['{"<element1>": <number_of_element1_atoms>, "<element2>": <number_of_element2_atoms>, ..., "<elementN>": <number_of_elementN_atoms>}'] --sampling-config-name=csp` to `generate.py`. 
+An example composition could be `--target_compositions=['{"Na": 1, "Cl": 1}']`.
 ### Fine-tuning on property data
 
 You can fine-tune the MatterGen base model using the following command.

